autoPyTorch/pipeline/components/training/trainer/base_trainer_choice.py

```diff
@@ -410,7 +410,7 @@ def _fit(self, X: Dict[str, Any], y: Any = None, **kwargs: Any) -> torch.nn.Modu
             if 'cuda' in X['device']:
                 torch.cuda.empty_cache()
 
-        if self.choice.use_stochastic_weight_averaging:
+        if self.choice.use_stochastic_weight_averaging and self.choice.swa_updated:
```

Comment:
When do you have to set `self.choice.swa_updated` back to `False`?
We need this change, right?

Reply:
I think for some pipeline it's enough that we change this once to True so to make sure the swa_model has meaning full weights. Especially in early stopping, there is a scenario when swa parameters were not updated. Leading to the model getting nearly random weights.